packages/main/test/samples/Tree.sample.html

```diff
@@ -0,0 +1,188 @@
+            <ui5-tree-item text="Tree 1.1.2"></ui5-tree-item>
+        </ui5-tree-item>
+    </ui5-tree-item>
+    <ui5-tree-item text="Tree 2" icon="copy">
+        <ui5-tree-item text="Tree 2.1">
+            <ui5-tree-item text="Tree 2.1.1"></ui5-tree-item>
+            <ui5-tree-item text="Tree 2.1.2">
+                <ui5-tree-item text="Tree 2.1.2.1"></ui5-tree-item>
+                <ui5-tree-item text="Tree 2.1.2.2"></ui5-tree-item>
+                <ui5-tree-item text="Tree 2.1.2.3"></ui5-tree-item>
+                <ui5-tree-item text="Tree 2.1.2.5"></ui5-tree-item>
+            </ui5-tree-item>
+        </ui5-tree-item>
+        <ui5-tree-item text="Tree 2.2"></ui5-tree-item>
+    </ui5-tree-item>
+
+    <ui5-tree-item expanded text="Tree 3 (no icon)">
+    </ui5-tree-item>
+</ui5-tree>
+	</xmp></pre>
+</section>
+
+<!-- dynamic -->
+<section>
+	<h3>Tree with dynamic content</h3>
+	<div class="snippet">
+        <ui5-tree id="treeDynamic" mode="None">
+            <ui5-tree-item text="Has pre-loaded children">
+                <ui5-tree-item text="Child 1"></ui5-tree-item>
+                <ui5-tree-item text="Child 2"></ui5-tree-item>
+            </ui5-tree-item>
+
+            <ui5-tree-item text="Has no children at all"></ui5-tree-item>
+
+            <ui5-tree-item id="dynamicNode" text="Has children, but not yet loaded" has-children></ui5-tree-item>
+        </ui5-tree>
+	</div>
+	<pre class="prettyprint lang-html"><xmp>
+<ui5-tree id="treeDynamic" mode="None">
+    <ui5-tree-item text="Has pre-loaded children">
+        <ui5-tree-item text="Child 1"></ui5-tree-item>
+        <ui5-tree-item text="Child 2"></ui5-tree-item>
+    </ui5-tree-item>
+
+    <ui5-tree-item text="Has no children at all"></ui5-tree-item>
+
+    <ui5-tree-item id="dynamicNode" text="Has children, but not yet loaded" has-children></ui5-tree-item>
+</ui5-tree>
+
+<script>
+	var dynamicTree = document.getElementById("treeDynamic");
+	dynamicTree.addEventListener("itemToggle", function(event) {
+		const item = event.detail.item; // get the node that is toggled
+
+		// Only for the dynamic node, and only when it's empty
+		if (item.id === "dynamicNode" && item.children.length === 0) {
+			dynamicTree.busy = true; // block the tree from the user
+			event.preventDefault(); // do not let the toggle button switch yet
+			setTimeout(function() {
+				const newItem = document.createElement("ui5-tree-item"); // Fetching from db....
+				newItem.text = "Node fetched from DB after 2 sec";
+				item.appendChild(newItem); // add the newly fetched node to the tree
+				item.toggle(); // now manually switch the toggle button
+				dynamicTree.busy = false; // unblock the tree
+			}, 2000);
+		}
+	});
+</script>
+	</xmp></pre>
+
+    <script>
+		var dynamicTree = document.getElementById("treeDynamic");
+		dynamicTree.addEventListener("itemToggle", function(event) {
+			const item = event.detail.item; // get the node that is toggled
```

Comment:
There are some "const" in the Tree.html and Tree.sample.html, maybe we should replace them with vars.